Cortexia-CH/VoTT
# Fix env variables in Docker

### Info
Environment variables used to be passed during build time and hardcoded in dockerfile, which prevented us from changing them after.
Now variables are in the docker environment. To allow users to change env variables, a script rewrites the value of the env variables in the application with the ones that are in the docker environment at runtime.
As the code of the application is chunked, the script is replacing the variables directly in the chunk files.
For each new env variable that might need to be modified without rebuilding the app, you will have to add a line in the `environment.ts` with a placeholder, and add a line in the `setup-env.sh` script to replace the placeholder.

### How to test
1. Complete your `.env` file as in the `.sample.env`
2. Change the `REACT_APP_API_URL` to any value
3. Change the `SUBDOMAIN` to `vott` and the `DOMAIN` to `local`, then check that you have `127.0.0.1 vott.local` in your `/etc/hosts` file
4. In your terminal, in `git-repos/VoTT`: `sourceenv`, then `make deploy-local`
5. Wait for a while, then when it's done, go to `vott.local`, inspect page and check the network tab => the `check-token` should try to access the URL that you set in your `.env` file
